cmake/CopyBuildProducts.cmake

```diff
@@ -1,5 +1,5 @@
 function(copy_build_products target)
-  set(valid_options "DEPENDENCY;DEPENDENCY_FILES;TARGET_LOCATION;CONDITION")
+  set(valid_options "DEPENDENCY;DEPENDENCY_FILES;TARGET_LOCATION;CONDITION;DROP_PATH")
```

Comment:
Could you explain what DROP_PATH is supposed to do? I'm having a bit of a hard time visualising it right now.

Reply:
`DROP_PATH` is probably a _really_ bad name but I could not come up with anything at the time.

Essentially it does two things:
- It drops the filepath of the file for the install target
(i.e. from `../../thirdparty/vorbis.dll` to `vorbis.dll`)
- It will adjust the written `copy_build_products` target to look up one additional directory
(from `../../thirdparty/vorbis.dll` to `../../../thirdparty/vorbis.dll`)

This was done for two reasons -
1. Using relative paths would cause `install` to fail:
>  CMake Error at src/NovelRT/cmake_install.cmake:60 (file):
>   file INSTALL cannot find "D:/repos/kennynovel/src/NovelRT/FLAC++.dll": No
>   error.


  2. Using absolute paths would cause `copy_build_products` to generate an invalid path when copying the Engine's build products

>   Copying required build products to EcsPipeline
>   Error copying file (if different) from "D:/repos/kennynovel/build/src/NovelRT/Debug/../../thirdparty/FLAC++.dll" to "
>   D:/repos/kennynovel/build/samples/EcsPipeline/Debug/../../thirdparty/FLAC++.dll".

Adding it as a configuration allows me to keep the resources separate (as they're working as is) yet also correct the issues that the `dll`s keep running into. I'm _200%_ open to suggestions or ideas on improving or correcting this as needed

Reply:
We could probably do another absolute path translation to fix point 2. This might be related to the fix for #387 - could you try that to see if it works?

Reply:
Sure thing! I'll give that a shot

Reply:
This wasn't working properly - I took a different approach and reverted the changes to the function.